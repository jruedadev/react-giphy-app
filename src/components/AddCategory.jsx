import { useState } from "react"
import { PropTypes } from "prop-types";

export const AddCategory = ({onNewCategory}) => {

    const [inputValue, setInputValue] = useState('Prueba');

    const onInputChange = ({target}) => {
        setInputValue(target.value);
    }

    const onSubmit = (e) => {
        e.preventDefault();

        const value = inputValue.trim();
        if (value.length < 2) return ;

        onNewCategory( value );
        setInputValue('');
    }

  return (
    <form onSubmit={onSubmit}>
        <input 
            type="text"  
            placeholder="Buscar Gif's" 
            value={ inputValue } 
            onChange={ onInputChange } 
        />
        <button type="submit" >Agregar</button>
    </form>
  )

}

AddCategory.propTypes = {
//    setCategories: PropTypes.Function.isRequired,
}