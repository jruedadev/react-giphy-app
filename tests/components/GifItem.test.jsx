import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('GifItemComponent', () => {

    const title = "Título";
    const url = "https://example.com/";
    
    test("debe hacer match con el snapshot", () => {
        const { container } = render(<GifItem title={title} url={url} />);
        expect(container).toMatchSnapshot();
    });
    
    test('Mostrar imagen con url y alt', () => {
        const { container } = render(<GifItem title={title} url={url} />);
        
        const {src, alt} = screen.getByRole('img');
        expect(alt).toBe(title);
        expect(src).toBe(url);
    });

    test('Mostrar el título en el componente', () => { 
        
        render(<GifItem title={title} url={url} />);
        expect(screen.getByText(title)).toBeTruthy();

     })
});