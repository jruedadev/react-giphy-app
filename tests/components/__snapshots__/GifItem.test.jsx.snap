// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifItemComponent debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Título"
      src="https://example.com"
    />
    <p>
      Título
    </p>
  </div>
</div>
`;
